### ISSUE TITLE
Exported jpeg Image not the same as whats seen in darkroom

### ISSUE THREAD

**Issue by chhil** [darktable-org/darktable#2852]
**Describe the bug**
After doing edits in darktable the exported jpeg image does not look like the final edited image.

**To Reproduce**
I have attached  an  image containing the history of darktable edits on the left and the the exported imaged on the right. Export done using shortcut ot through lightable view export button has same outcome. Various grades of jpeg from 70 to 100 for exported jpegs were used but that did not alter behavior.

**Expected behavior**
The iexported image should look exactly like the final edited one.
**Screenshots**

**Platform (please complete the following information):**
 - OS: [Win 10]
 - Version [darktable-2.7.0+1612~g9e212dc9d-win64] I built this on my own.

**Additional context**
I think its losing the retouch edits in the export. 
![image](https://user-images.githubusercontent.com/1119439/62890839-e8fec980-bd61-11e9-98e9-8a753ce07327.png)

**Comment by grumpazoid** [comment]
Do things improve if you export at a lower resolution such as 2048? See my bug report 2845.

**Comment by chhil** [comment]
@grumpazoid 
I tried 2048. No change. I don't have haze removal enabled either, though I have seen issues with haze removal and retouch module messing up the output.

**Comment by grumpazoid** [comment]
@chhil Looks like you have a different issue to me then. Maybe related somehow?

**Comment by kstar** [comment]
I have a similar problem with darktable 2.6.2 on Arch Linux.
In fact, no matter what format I export to (even 16-bit formats), the results are the same -- I get posterization of color. To confirm that this is not a image viewer issue, I reopened the 16-bit file in darktable and get the same posterization in darktable.

Input image:
![IMG-0265](https://user-images.githubusercontent.com/351472/63208554-583e2b80-c08a-11e9-95bc-ab6d6be8bc0d.JPG)

Edit .xmp file (renamed to .xmp.txt in order to upload to github)
[IMG-0265.JPG.xmp.txt](https://github.com/darktable-org/darktable/files/3511772/IMG-0265.JPG.xmp.txt)

Output (this particular file was written with a slightly older version of darktable, but the results with the 2.6.2 are similar):
![HalfDome](https://user-images.githubusercontent.com/351472/63208573-a7845c00-c08a-11e9-9904-12cb802f030f.jpeg)

Since darktable is able to display the file perfectly on a standard laptop monitor display, I imagine it should be possible to convert it correctly at least to a 16-bit format...

**Comment by chhil** [comment]
The file was edited in Darktable 2.7.xxxxx. Since export isnt working in some scenarios where I have used the retouch module, I did the following to work around it.

I opened the same raw through gimp which opened it in darktable. I closed darktable and the same image without issues came up in gimp. From there I exported it. 

This might help understand the issue as I feel in both cases an export is taking place but in one case its failing.
